### PR TITLE
fix(autofix): Batching chunk inserts

### PR DESCRIPTION
**Pull request by jennmueng**
Should fix [TIMESERIES-ANALYSIS-SERVICE-2S](https://sentry.sentry.io/issues/5029600041/)

### CODE REVIEW

File: src/seer/utils.py [hunk 1]
```diff
@@ -2,6 +2,9 @@
 import json
 import weakref
 from enum import Enum
+from typing import Sequence
+
+from sqlalchemy.orm import DeclarativeBase, Session
```
Comment (trillville) [inline src/seer/utils.py:7]: shouldnt this be from `seer.db`?

File: src/seer/utils.py [hunk 2]
```diff
@@ -34,3 +37,14 @@ def default(self, obj):
 
 def json_dumps(data, **kwargs) -> str:
     return json.dumps(data, cls=SeerJSONEncoder, **kwargs)
+
+
+def batch_save_to_db(session: Session, data: Sequence[DeclarativeBase], batch_size: int = 512):
+    """
+    Save a list of data to the database in batches. Flushes the session after each batch.
+    """
+    for i in range(0, len(data), batch_size):
```
Comment (trillville) [inline src/seer/utils.py:46]: i think you can wrap this entire operation in a transaction like this:

```
with session.begin():
            for i in range(0, len(data), batch_size):
                session.bulk_save_objects(data[i : i + batch_size])

                session.flush()
```

Comment (jennmueng) [reply src/seer/utils.py:46]: it's supposed to be in the same transaction as the parent not its own

Comment (jennmueng) [reply src/seer/utils.py:46]: lemme add comment

Comment (trillville) [reply src/seer/utils.py:47]: ah perfect i see

**Review by trillville — APPROVED** [review]

**Pull request merged**